osu.Game/Rulesets/Mods/OverridableBindable.cs

```diff
@@ -0,0 +1,165 @@
+        public Bindable<T> FinalValue => finalValue;
+
+        /// <summary>
+        /// Whether the final bindable should be bound to <see cref="CustomValue"/>, otherwise will be immutably bound to <see cref="BaseValue"/>.
+        /// </summary>
+        public Bindable<bool> HasCustomValue { get; } = new BindableBool();
+
+        private readonly BindableNumber<T> immutableBaseValue = new BindableNumber<T>();
+        private readonly BindableNumber<T> customValue = new BindableNumber<T>();
+        private readonly BindableNumber<T> finalValue = new BindableNumber<T>();
+
+        private readonly LeasedBindable<T> baseValue;
+
+        public OverridableBindable(T? defaultValue = null, T? minValue = null, T? maxValue = null, T? precision = null)
+        {
+            if (defaultValue != null)
+            {
+                immutableBaseValue.Value = customValue.Value = finalValue.Value = defaultValue.Value;
+                immutableBaseValue.Default = customValue.Default = finalValue.Default = defaultValue.Value;
+            }
+
+            if (minValue != null)
+                immutableBaseValue.MinValue = customValue.MinValue = finalValue.MinValue = minValue.Value;
+
+            if (maxValue != null)
+                immutableBaseValue.MaxValue = customValue.MaxValue = finalValue.MaxValue = maxValue.Value;
+
+            if (precision != null)
+                immutableBaseValue.Precision = customValue.Precision = finalValue.Precision = precision.Value;
+
+            // this lease is never returned on purpose.
+            // the leased bindable is exposed publicly to be mutable from outside,
+            // while the inner one is immutable, so that updateFinalValue() can bind to it if HasCustomValue = false,
+            // therefore ensuring that FinalValue is disabled.
+            baseValue = immutableBaseValue.BeginLease(false);
+
+            HasCustomValue.BindValueChanged(_ => updateFinalValue(), true);
+        }
+
+        /// <summary>
+        /// Parameterless constructor for serialization.
+        /// </summary>
+        [UsedImplicitly]
+        private OverridableBindable()
+            : this(null)
+        {
+        }
+
+        /// <summary>
+        /// Parses an input into <see cref="CustomValue"/> and sets <see cref="HasCustomValue"/> accordingly.
+        /// </summary>
+        /// <param name="input">The input which is to be parsed.</param>
+        public void Parse(object input)
+        {
+            switch (input)
+            {
+                case null:
+                    HasCustomValue.Value = false;
+                    return;
+
+                case OverridableBindable<T> setting:
+                    if (setting.HasCustomValue.Value)
+                    {
+                        CustomValue.Value = setting.CustomValue.Value;
+                        HasCustomValue.Value = true;
+                    }
+                    else
+                        HasCustomValue.Value = false;
```

Comment:
can be simplified to
```suggestion
					HasCustomValue.Value = setting.CustomValue.Value;
                    if (setting.HasCustomValue.Value)
                        CustomValue.Value = setting.CustomValue.Value;
```

Reply:
intentionally did it this way such that `FinalValue` binds to `CustomValue` (`HasCustomValue = true`) after the value is parsed.

Reply:
then reverse the operation order? with a comment, obviously. no idea why there was no comment in the first place if this was relying on weird operation ordering.

Reply:
thought it was quite clear, not say "I have custom value" unless I actually do have a valid one, so you mean like this:
```suggestion
                    HasCustomValue.Value = false;

                    if (setting.HasCustomValue.Value)
                    {
                        CustomValue.Value = setting.CustomValue.Value;
                        HasCustomValue.Value = true;
					}
```

Reply:
why is the assignment to `HasCustomValue` under a branch at all if you're just copying the value, is what i'm getting at

Reply:
Resolved in https://github.com/ppy/osu/pull/11387/commits/831e11dbf2a608bbbc5f061b846cc4a5caedcdf7 (should've been in a separate commit but /shrug)